frappe/__init__.py

```diff
@@ -291,8 +291,9 @@ def connect(
 	local.db = get_db(
 		host=local.conf.db_host,
 		port=local.conf.db_port,
-		user=local.conf.db_user or db_name,
-		password=None,
+		user=local.conf.db_user or db_name or local.conf.db_name,
```

Comment:
Postgres:
![image](https://github.com/frappe/frappe/assets/7548295/34cebf09-074a-4d5e-a560-c0a8d5696221)

---

Mariadb:
![image](https://github.com/frappe/frappe/assets/7548295/a1537033-e83e-4a02-a5ec-fcb0a6887e14)

Reply:
-> https://github.com/frappe/frappe/pull/24428

Reply:
This may have been fixed by https://github.com/frappe/frappe/pull/24434, but I'd prefer to do a follow up PR and remove it rather than further doctoring with this PR, if possible :pray:

Reply:
Yeah makes sense, will merge for now, will try to figure out the best way to clean it up
Basically the additional fallback is just for cases where someone has called frappe.connect() without initing and without passing a site name, right?